packages/relay/src/lib/eth.ts

```diff
@@ -503,6 +503,13 @@ export class EthImpl implements Eth {
     const requestIdPrefix = formatRequestIdMessage(requestId);
     this.logger.trace(`${requestIdPrefix} getBalance(account=${account}, blockNumberOrTag=${blockNumberOrTag})`);
 
+    if (!EthImpl.blockTagIsLatestOrPending(blockNumberOrTag)) {
+      const latestBlock = await this.blockNumber(requestId);
+      if (latestBlock === blockNumberOrTag) {
+        blockNumberOrTag = latestBlock
```

Comment:
ALso if not already covered please add a test case in `eth.spec.ts` to make sure the call made to the mirror node is as expected.